packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -207,6 +207,36 @@ mixin class ChangeNotifier implements Listenable {
   @protected
   bool get hasListeners => _count > 0;
 
+  /// Dispatches event of object creation to [MemoryAllocations.instance].
+  ///
+  /// If the event was already dispatched or [kFlutterMemoryAllocationsEnabled]
+  /// is false, the method is noop.
+  ///
+  /// Tools like leak_tracker use the event of object creation to help
+  /// developers identify the owner of the object, for troubleshooting purposes,
+  /// by taking stack trace at the moment of the event.
+  ///
+  /// But, as [ChangeNotifier] is mixin, it does not have its own constructor. So, it
+  /// communicates object creation in first `addListener`, that results
+  /// in the stack trace pointing to `addListener`, not to constructor.
+  ///
+  /// To make debugging easier, invoke [ChangeNotifier.maybeDispatchObjectCreation]
```

Comment:
Let's add to the documentation that it should be guarded by `if (kFlutterMemoryAllocationsEnabled) ...`